src/Microsoft.IdentityModel.JsonWebTokens/JsonWebTokenHandler.CreateToken.cs

```diff
@@ -991,7 +995,7 @@ internal static void WriteJwsHeader(
             else
             {
                 writer.WriteString(JwtHeaderUtf8Bytes.Alg, signingCredentials.Algorithm);
-                if (signingCredentials.Key.KeyId != null)
+                if (includeKidInHeader && signingCredentials.Key.KeyId != null)
```

Comment:
Since the x5t is also a type of 'kid', we should skip writing out the x5t if includeKidInHeader is false.

Reply:
Does the same apply to `alg`, then?